src/globus_sdk/transport/requests.py

```diff
@@ -335,15 +340,27 @@ def request(
         resp: requests.Response | None = None
         req = self._encode(method, url, query_params, data, headers, encoding)
         checker = RetryCheckRunner(self.retry_checks)
+
+        # Always ensure we have caller_info
+        if caller_info is None:
+            caller_info = RequestCallerInfo(authorizer=authorizer)
+        elif authorizer is not None and caller_info.authorizer is not authorizer:
+            # Warn if both parameters are provided but differ
+            log.warning(
+                "Both 'caller_info' and 'authorizer' parameters provided to "
+                "request(). Using authorizer from caller_info and ignoring "
+                "standalone authorizer parameter."
+            )
```

Comment:
I don't know if this was a TAC item but I would prefer for this to be an exception, not a warning.

My reasoning is that no existing code will be affected by this new `caller_info` parameter. However, developers that make a deliberate choice to use the `caller_info` parameter are opting into a new contract. Therefore, if they pass both in, I think that we're safe to raise an exception and explicitly say "don't pass both".

It's a clearer contract than a warning, I think.

Reply:
I've made updates that remove this code, so this comment no longer applies.